app/Server/src/SERVERAPI/Pages/MiniApps/SoilTestConverter.cshtml.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Agri.CalculateService;
+using Agri.Data;
+using Agri.Models.Configuration;
+using Agri.Models.Farm;
+using FluentValidation;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SERVERAPI.Pages.MiniApps
+{
+    public class SoilTestConverter : PageModel
+    {
+        private readonly IMediator _mediator;
+
+        [BindProperty]
+        public ConverterQuery Data { get; set; }
+
+        [BindProperty]
+        public ResultModel Result { get; set; }
+
+        public SoilTestConverter(IMediator mediator) => _mediator = mediator;
+
+        public async Task OnGetAsync()
+        {
+            Data = await _mediator.Send(new Query { PopulatedData = new ConverterQuery() });
+            Result = new ResultModel();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            if (ModelState.IsValid)
+            {
+                //Send the Data to Receive conversion result
+                Result = await _mediator.Send(Data);
+                ModelState.Remove("Result.KelownaConversion");
+            }
+            Data = await _mediator.Send(new Query { PopulatedData = Data });
+            return Page();
+        }
+
+        public class Query : IRequest<ConverterQuery>
+        {
+            public ConverterQuery PopulatedData { get; set; }
+        }
+
+        public class ConverterQuery : IRequest<ResultModel>
+        {
+            public decimal pH { get; set; }
+            public decimal phosphorous { get; set; }
+            public List<SelectListItem> laboratoryOptions { get; set; }
+            public string selLaboratoryOption { get; set; }
```

Comment:
Please when possible ensure that all Properties start with a Capital letter for consistency.